### PR TITLE
[HS-1071657] Migrate cart and view product data layer events to GA4

### PR DESCRIPTION
**Pull request by canac**
Migrate add_to_cart, remove_from_cart, and view_item events to GA4.

https://secure.helpscout.net/conversation/2457822430/1071657?viewId=7296729

### CODE REVIEW

File: src/app/analytics/analytics.factory.js
Comment (wrandall22) [inline src/app/analytics/analytics.factory.js:256]: I would be curious how this is working, and not needing to be `AMOUNT`. I know the old code is doing it, but maybe it is broken?

Comment (wrandall22) [reply src/app/analytics/analytics.factory.js:256]: I guess it wouldn't be broken, otherwise the cart would not show the data since the `cart.tpl.html` is using lowercase.

Comment (wrandall22) [reply src/app/analytics/analytics.factory.js:256]: I see where the transformation is, in `cart.service.js` `handleCartResponse()`

**Comment by wrandall22** [comment]
I'd prefer the first 2 commits to be squashed together before merging.

**Comment by canac** [comment]
Will requested that `testing_transaction` be explicitly set to `false` instead of being missing.

We should consider sending a boolean instead of the strings `"true"` and `"false"` like branded checkout does, but I don't want to change that yet until we can prove that it won't break anything.

**Review by wrandall22 — APPROVED** [review]

**Comment by canac** [comment]
According to Will, this change is not high urgency and wait until after the code freeze to be merged.

File: src/app/analytics/analytics.factory.js
Comment (dr-bizz) [inline src/app/analytics/analytics.factory.js:451]: Will there ever be a `view_item` which has a value? If no, should we have it this object, or do the analytics guys need it?

**Review by dr-bizz — COMMENTED** [review]
Everything looks great! Thanks for improving my code with the `testingTransaction`

File: src/app/analytics/analytics.factory.js
Comment (canac) [reply src/app/analytics/analytics.factory.js:451]: Good question. `view_item` won't ever have a `value` amount at this stage because the user hasn't selected the gift amount or frequency yet. When I implemented the GA4 spec for branded checkout, the spec docs said, "If no value exists for an attribute in the dataLayer, please pass JavaScript `undefined` as the value." That's why I'm passing `undefined` instead of omitting that field.

**Pull request merged**
